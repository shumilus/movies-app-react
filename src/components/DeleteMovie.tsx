import * as React from 'react';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import { createUseStyles } from 'react-jss';
import Button from './Button';

interface MovieCardMenuProps {
  isOpen: boolean;
  outsideClick: () => void;
  confirmClick: () => void;
}

const useStyles = createUseStyles({
  content: {
    padding: '60px 80px',
    color: '#FFFFFF',
  },
  title: {
    fontWeight: '300',
    fontSize: '40px',
    lineHeight: '49px',
    letterSpacing: '1px',
    textTransform: 'uppercase',
    marginTop: '0',
  },
  subTitle: {
    fontWeight: '400',
    fontSize: '20px',
    lineHeight: '24px',
    marginBottom: '50px',
  },
});

export default function DeleteMovie(props: MovieCardMenuProps) {
  const classes = useStyles();

  return (
    <Dialog
      open={props.isOpen}
      onClose={props.outsideClick}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <button className="close-button" onClick={props.outsideClick}></button>

      <div className={classes.content}>
        <h2 className={classes.title}>delete movie</h2>
        <p className={classes.subTitle}>Are you sure you want to delete this movie?</p>

        <DialogActions>
          <Button classes='primary-button' title='confirm' handleClick={props.confirmClick}></Button>
        </DialogActions>
      </div>
    </Dialog>
  );
}
