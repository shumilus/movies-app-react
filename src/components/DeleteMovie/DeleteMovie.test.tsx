import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import DeleteMovie from './DeleteMovie';

describe('DeleteMovie', () => {
  it('should match DeleteMovie snapshot', () => {
    const { asFragment } = render(<DeleteMovie isOpen={true} onOutsideClick={() => {}} onConfirmClick={() => {}}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger handleConfirmClick on confirm button click', () => {
    const handleConfirmClick = jest.fn();
    const handleOutsideClick = jest.fn();

    render(<DeleteMovie isOpen={true} onConfirmClick={handleConfirmClick} onOutsideClick={handleOutsideClick}/>);
    userEvent.click(screen.getByRole('button', { name: /confirm-btn/i }));

    expect(handleConfirmClick).toHaveBeenCalledTimes(1);
  });

  it('should trigger handleOutsideClick on close button click', () => {
    const handleConfirmClick = jest.fn();
    const handleOutsideClick = jest.fn();

    render(<DeleteMovie isOpen={true} onConfirmClick={handleConfirmClick} onOutsideClick={handleOutsideClick}/>);
    userEvent.click(screen.getByRole('button', { name: /close-btn/i }));

    expect(handleOutsideClick).toHaveBeenCalledTimes(1);
  });
});
