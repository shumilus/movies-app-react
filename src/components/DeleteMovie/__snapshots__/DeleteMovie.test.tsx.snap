// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovie should match DeleteMovie snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
